artifact-manifest/artifact-manifest.md

```diff
@@ -0,0 +1,484 @@
+
+The following command would copy the `mysql:8` image from docker hub to the acme-rockets registry. The CLI _could_ be run within the source or target cloud eliminating the download/upload network hops.
+
+```bash
+oci-reg copy \
+  --source docker.io/mysql:8 \
+  --target registry.acme-rockets.io/base-artifacts/mysql:8
+```
+
+The `oci-reg copy` command would:
+
+- assure the manifest and layer/blob digests remain the same
+- copy any artifacts that are dependent on the source artifact-manifest, persisting them in the target registry. These _could_ include Notary v2 signatures, SBoMs, GPL source or other referenced artifacts.
+
+### Reference Artifacts
+
+There are a set of artifact types that declare references to other artifacts that may, or may not be stored in the same registry. The reference is important to note, indicating copying to be capable between environments, as well as generalized validation scenarios.
+
+#### Helm Reference
+
+![mysql image copy](./media/wordpress-helm-chart-copy.svg)
+
+In the above scenario, a helm chart is copied from a public registry to the ACME Rockets registry.  The `wordpress-chart:v5` is represented as an `application/vnd.oci.artifact.manifest.v1+json`. The `wordpress-chart:v5` helm chart references the `wordpress:v5` image and the `mysql:8` image. All three artifacts have Notary v2 signatures attesting to their authenticity.
+
+As the copy is initiated, the `oci.artifact.manifest` of the `wordpress-chart:v5` is evaluated. As the chart references the same version (digest) of the `mysql:8` image already in theACME Rockets registry, the copy skips duplicating the content and moves to copying the `wordpress:v5` image, the `wordpress-chart:v5` and their associated signatures.
+
+In this case, the images referenced in the chart existed in the source registry. However, this may not be the case in all scenarios. A helm chart may be acquired from a different location, referencing images from docker hub. This deferred validation is left to client tools, enabling common package management resolution scenarios.
+
+To support the loose references between artifacts, a `references` collection is added to the `oci.artifact.manifest`:
+
+**A `wordpress-chart:v5` Helm Chart example:**
+
+```json
+{
+  "mediaType": "application/vnd.oci.artifact.manifest.v1+json",
+  "artifactType": "application/vnd.cncf.helm.v3",
+  "config": {
+    "mediaType": "application/vnd.cncf.helm.config.v1+json",
+    "digest": "sha256:b5b2b2c507a0944348e0303114d8d93aaaa081732b86451d9bce1f432a537bc7",
+    "size": 0
+  },
+  "blobs": [
+    {
+      "mediaType": "application/vnd.cncf.helm.chart.v1+tar",
+      "digest": "sha256:9834876dcfb05cb167a5c24953eba58c4ac89b1adf57f28f2f9d09af107ee8f0",
+      "size": 32654
+    },
+    {
+      "mediaType": "application/vnd.cncf.helm.values.v1+yaml",
+      "digest": "sha256:3c3a4604a545cdc127456d94e421cd355bca5b528f4a9c1905b15da2eb4a4c6b",
+      "size": 16724
+    }
+  ],
+  "manifests": [
```

Comment:
Does it assume the referenced manifest is persisted in the same registry?
This breaks a very common usage pattern that the referenced images are set in `values.yaml` of a chart, and the deployer of a chart will set the images why he issues `helm install`

Reply:
The idea is _they can_ be stored in the same registry and the registry can understand the references, genericially. This enables `oci-reg copy` scenarios and optional `oci-reg delete --with-references` scenarios. But, you're correct that this information is duplicated in the helm chart, as this `artifact.manifest` didn't (still doesn't) exist, yet. If/when this `artifact.manifest` gets adopted, would a combination of this manifest reference, along with [oci-reg.config](https://github.com/notaryproject/nv2/discussions/31) enable a helm chart to evolve, where the chart references the thing in the manifest. In theory, this would make it far easier to move charts across registries without having to change the chart.